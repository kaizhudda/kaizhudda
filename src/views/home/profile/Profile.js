import React from 'react';
import Title from '../../../sharedComponents/title';
import Subtitle from '../../../sharedComponents/subtitle';
import SecondaryTitle from '../../../sharedComponents/secondary-title';
import Experience from './experience';
import reduxLogo from '../../../assets/logos/redux.png';
import webpackLogo from '../../../assets/logos/webpack.svg';
import sassLogo from '../../../assets/logos/sass.png';
import enzymeLogo from '../../../assets/logos/enzyme.png';
import jestLogo from '../../../assets/logos/jest-svg.svg';
import './Profile.scss';

const Profile = () => {
  const skillsOne = (
    <>
      <span className="icon-container">
        <i className="fab fa-react" />
        <span className="tech-title">React</span>
      </span>
      <span className="icon-container">
        <img src={reduxLogo} className="redux-icon" alt="Redux"/>
        <span className="tech-title">Redux</span>
      </span>
      <span className="icon-container">
        <img src={sassLogo} className="redux-icon" alt="SASS"/>
        <span className="tech-title">SASS</span>
      </span> 
      <span className="icon-container">
        <i className="fab fa-js" />
        <span className="tech-title">ES6</span>
      </span>
      <span className="icon-container">
        <i className="fab fa-html5" />
        <span className="tech-title">HTML5</span>
      </span>
      <span className="icon-container">
        <i className="fab fa-css3-alt" />
        <span className="tech-title">CSS3</span>
      </span>  
      <span className="icon-container">
        <img src={webpackLogo} className="webpack-icon" alt="Webpack"/>
        <span className="tech-title">Webpack</span>
      </span>    
      <span className="icon-container">
        <img src={jestLogo} className="redux-icon" alt="Jest"/>
        <span className="tech-title">Jest</span>
      </span>  
      <span className="icon-container">
        <img src={enzymeLogo} className="redux-icon" alt="Enzyme"/>
        <span className="tech-title">Enzyme</span>
      </span>  
      <span className="icon-container">
        <i className="fab fa-git-alt" />
        <span className="tech-title">Git</span>
      </span>  
    </>
  );

  const skillsTwo = (
    <>
      <i className="fab fa-react" />
      <i className="fab fa-js" />
      <i className="fab fa-html5" />
      <i className="fab fa-css3-alt" />
    </>
  );

  const skillsThree = (
    <>
      <i className="fab fa-react" />
      <i className="fab fa-js" />
      <i className="fab fa-html5" />
      <i className="fab fa-css3-alt" />
    </>
  );

  const skillsFour = (
    <>
      <i className="fab fa-react" />
      <i className="fab fa-js" />
      <i className="fab fa-html5" />
      <i className="fab fa-css3-alt" />
    </>
  );

  return (
    <div className="Profile row center-xs" id="experience-section">
      <Title>Work Experience</Title>
      <Subtitle>"I am a great believer in luck, and I find the harder I work the more I have of it." - Stephen Leacock</Subtitle>
      <SecondaryTitle>Career</SecondaryTitle>
      <Experience
        title={'Front End Developer'}
        companyName={'Spark New Zealand'}
        period={'February 2019 - Present'}
        location={'Auckland, New Zealand'}
        companyLink={'https://spark.co.nz'}
        skills={skillsOne}>
        •	Developed and maintained 5 different customer facing web journeys on Sparks corporate website
        <br />
        •	Built out Spark’s reusable React component library which heavily reduced code duplication
        <br />
        •	Wrote and refactored unit tests to improve the reliability of the code base, using Jest and Enzyme
        <br />
        •	Defined and implemented best practices and coding standards for the organization to deliver high quality code
        <br />
        •	Worked closely with UI/UX designers to ensure design is translated correctly
       <br />
        •	Worked with AEM developers to seamlessly integrate React.js and add content authoring across journey’s
        <br />
        •	Mentored developers / Pair programming
        <br />
        •	Managed 2 weekly sprints for the frontend team with Azure DevOps
        </Experience>

      <Experience
        title={'Front End Developer'}
        companyName={'Genesis Energy'}
        period={'March 2017 – February 2019 (2 years)'}
        location={'Auckland, New Zealand'}
        companyLink={'https://genesisenergy.co.nz'}
        skills={'React, Redux, ES6/7 JavaScript, HTML5, CSS3/SCSS, Jest, Enzyme, Node.js, Git, Azure DevOps'}>
        •	Developed & maintained the Energy Management and Customer Self-Service web app from scratch using React, Redux, HTML5, CSS3/SASS, and JavaScript ES6
            <br />
        •	Code reviewed other developers code to ensure code quality and standards are met
            <br />
        •	Created/maintained test scripts with Jest and Enzyme reducing future defects and bugs
            <br />
        •	Worked closely with UI/UX designers to ensure design is translated correctly
            <br />
        •	Worked with Node.js/Express.js/MongoDB backend with Docker microservices to support the Front End
            <br />
        •	Contributed to development of APIs, including, electricity/LPG/natural gas usage data and user data
            <br />
        •	Tracked and managed 2 weekly sprints with JIRA
        </Experience>

      <Experience
        title={'Microsoft Intern'}
        companyName={'Genesis Energy'}
        period={'December 2016 – February 2017 (3 months)'}
        location={'Auckland, New Zealand'}
        companyLink={'https://genesisenergy.co.nz'}
        skills={'HTML5, CSS/SCSS, C#, Visual Studio, JavaScript, VSTS, SQL, Azure'}>
        •	Developed an energy monitoring and management dashboard prototype, using HTML5, CSS3, JavaScript, C#, Visual Studio and Bootstrap
          <br />
        •	Developed an LPG monitoring prototype using Visual Studio in 3 days, saving the company $25K
          <br />
        •	Integrated .NET C# JSON APIs with the Front End
          <br />
        •	Continuously deployed web projects using Microsoft Azure
          <br />
        •	Wrote web automation test scripts in C# and conducted cross browser testing
          <br />
        •	Extracted information from SQL database, to build managerial reports on key user metrics
        </Experience>

      <Experience
        title={'Test Analyst Intern'}
        companyName={'Foster Moore'}
        period={'November 2015 – February 2016 (4 months)'}
        location={'Auckland, New Zealand'}
        companyLink={'https://fostermoore.co.nz'}
        skills={'Selenium, RobotFramework, Python, Jenkins'}>
        •	Converted legacy web automation test scripts using Robot Framework, Selenium & Python
          <br />
        •	Troubleshooted and optimised automation web tests and CI/CD pipelines
          <br />
        •	Continuously deployed test scripts with Jenkins
          <br />
        •	Customised Jenkins to run and view test results of the latest deployment automatically
        </Experience>

      <Experience
        title={'Software Engineer Intern'}
        companyName={'Orion Health'}
        period={'November 2014 – February 2015 (4 months)'}
        location={'Auckland, New Zealand'}
        companyLink={'https://orionhealth.com'}
        skills={'HTML5, CSS3, Javascript, Cassandra (No SQL), Python3'}>
        •	Converted legacy web automation test scripts using Robot Framework, Selenium & Python
          <br />
        •	Troubleshooted and optimised automation web tests and CI/CD pipelines
          <br />
        •	Continuously deployed test scripts with Jenkins
          <br />
        •	Customised Jenkins to run and view test results of the latest deployment automatically
        </Experience>

      <SecondaryTitle>Education</SecondaryTitle>
      <Experience
        title={'BSc Computer Science & Information Systems'}
        companyName={'University of Auckland'}
        period={'2014 – 2017'}
        location={'Auckland, New Zealand'}
        companyLink={'https://auckland.ac.nz'}
        education>
      </Experience>

    </div>
  );
}

export default Profile;
