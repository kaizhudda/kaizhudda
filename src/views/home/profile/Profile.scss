@import '../../../styles/variables';
@import '../../../styles/media-queries';

.Profile {
  background: #F3EFE0;
  color: #727878;
  font-family: $font-body;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 16px 16px 48px 16px;
}