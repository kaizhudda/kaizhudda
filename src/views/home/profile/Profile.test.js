import React from 'react';
import { shallow } from 'enzyme';
import Profile from './Profile';

let wrapped = shallow(<Profile />); 

describe('Profile', () => {
  it('should render the Profile Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should render the Title Component', () => {
    expect(wrapped.find('Title').exists()).toBe(true);
  });

  it('should render the subtitle component', () => {
    expect(wrapped.find('Subtitle').exists()).toBe(true);    
  });

  it('should render the secondary Title component', () => {
    expect(wrapped.find('SecondaryTitle').exists()).toBe(true);    
  });

  it('should render the Experience component', () => {
    expect(wrapped.find('Experience').length).toBe(6);    
  });
});
