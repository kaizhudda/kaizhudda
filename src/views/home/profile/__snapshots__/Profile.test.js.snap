// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render the Profile Component correctly 1`] = `
<div
  className="Profile row center-xs"
  id="experience-section"
>
  <Title>
    Work Experience
  </Title>
  <Subtitle>
    "I am a great believer in luck, and I find the harder I work the more I have of it." - Stephen Leacock
  </Subtitle>
  <SecondaryTitle>
    Career
  </SecondaryTitle>
  <Experience
    companyLink="https://spark.co.nz"
    companyName="Spark New Zealand"
    location="Auckland, New Zealand"
    period="February 2019 - Present"
    skills="React, Redux, ES6/7 JavaScript, Webpack 4, HTML5, CSS3/SCSS, Node.js, Jest, Enzyme, Styled components, Git, Azure DevOps"
    title="Front-End React Developer (Mid-Level)"
  >
    • Developed and maintained 5 different customer facing web journeys on Sparks corporate website
    <br />
    • Built out Spark’s reusable React component library which heavily reduced code duplication
    <br />
    • Wrote and refactored unit tests to improve the reliability of the code base, using Jest and Enzyme
    <br />
    • Defined and implemented best practices and coding standards for the organization to deliver high quality code
    <br />
    • Worked closely with UI/UX designers to ensure design is translated correctly
    <br />
    • Worked with AEM developers to seamlessly integrate React.js and add content authoring across journey’s
    <br />
    • Mentored developers / Pair programming
    <br />
    • Managed 2 weekly sprints for the frontend team with Azure DevOps
  </Experience>
  <Experience
    companyLink="https://genesisenergy.co.nz"
    companyName="Genesis Energy"
    location="Auckland, New Zealand"
    period="March 2017 – February 2019 (2 years)"
    skills="React, Redux, ES6/7 JavaScript, HTML5, CSS3/SCSS, Jest, Enzyme, Node.js, Git, Azure DevOps"
    title="Front End Developer"
  >
    • Developed & maintained the Energy Management and Customer Self-Service web app from scratch using React, Redux, HTML5, CSS3/SASS, and JavaScript ES6
    <br />
    • Code reviewed other developers code to ensure code quality and standards are met
    <br />
    • Created/maintained test scripts with Jest and Enzyme reducing future defects and bugs
    <br />
    • Worked closely with UI/UX designers to ensure design is translated correctly
    <br />
    • Worked with Node.js/Express.js/MongoDB backend with Docker microservices to support the Front End
    <br />
    • Contributed to development of APIs, including, electricity/LPG/natural gas usage data and user data
    <br />
    • Tracked and managed 2 weekly sprints with JIRA
  </Experience>
  <Experience
    companyLink="https://genesisenergy.co.nz"
    companyName="Genesis Energy"
    location="Auckland, New Zealand"
    period="December 2016 – February 2017 (3 months)"
    skills="HTML5, CSS/SCSS, C#, Visual Studio, JavaScript, VSTS, SQL, Azure"
    title="Microsoft Intern"
  >
    • Developed an energy monitoring and management dashboard prototype, using HTML5, CSS3, JavaScript, C#, Visual Studio and Bootstrap
    <br />
    • Developed an LPG monitoring prototype using Visual Studio in 3 days, saving the company $25K
    <br />
    • Integrated .NET C# JSON APIs with the Front End
    <br />
    • Continuously deployed web projects using Microsoft Azure
    <br />
    • Wrote web automation test scripts in C# and conducted cross browser testing
    <br />
    • Extracted information from SQL database, to build managerial reports on key user metrics
  </Experience>
  <Experience
    companyLink="https://fostermoore.co.nz"
    companyName="Foster Moore"
    location="Auckland, New Zealand"
    period="November 2015 – February 2016 (4 months)"
    skills="Selenium, RobotFramework, Python, Jenkins"
    title="Test Analyst Intern"
  >
    • Converted legacy web automation test scripts using Robot Framework, Selenium & Python
    <br />
    • Troubleshooted and optimised automation web tests and CI/CD pipelines
    <br />
    • Continuously deployed test scripts with Jenkins
    <br />
    • Customised Jenkins to run and view test results of the latest deployment automatically
  </Experience>
  <Experience
    companyLink="https://orionhealth.com"
    companyName="Orion Health"
    location="Auckland, New Zealand"
    period="November 2014 – February 2015 (4 months)"
    skills="HTML5, CSS3, Javascript, Cassandra (No SQL), Python3"
    title="Software Engineer Intern"
  >
    • Converted legacy web automation test scripts using Robot Framework, Selenium & Python
    <br />
    • Troubleshooted and optimised automation web tests and CI/CD pipelines
    <br />
    • Continuously deployed test scripts with Jenkins
    <br />
    • Customised Jenkins to run and view test results of the latest deployment automatically
  </Experience>
  <SecondaryTitle>
    Education
  </SecondaryTitle>
  <Experience
    companyLink="https://auckland.ac.nz"
    companyName="University of Auckland"
    education={true}
    location="Auckland, New Zealand"
    period="2014 – 2017"
    title="BSc Computer Science & Information Systems"
  />
</div>
`;
